api.py

```diff
@@ -1,68 +1,114 @@
 import hug
+import html
 import falcon
+import logging
 import maxminddb
 import time
-import logging
+import os
+
+import geoip2.webservice
+from geoip2.errors import GeoIP2Error
+
+
+GEOIP2_USER_ID = os.getenv('GEOIP2_USER_ID', '')
+GEOIP2_LICENSE_KEY = os.getenv('GEOIP2_LICENSE_KEY', '')
+GEOIP_SERVICE = 2
+
+result = {
+    'ip' : '',
+    'continent_code' : '',
+    'continent_name' : '',
+    'country_code' : '',
+    'country_name' : '',
+    'region_code' : '',
+    'region_name' : '',
+    'city' : '',
+    'zip' : '',
+    'latitude' : '',
+    'longitude' : '',
+    'location' : {
+        'geonameid' : '',
+        'time_zone' : ''
+    }
+}
+
 
 class GeoIP:
-    def __init__(self, reader):
+    def __init__(self, reader: None):
         self.reader = reader
 
     @hug.object.get(urls='/geoip/{ipv4}')
     def get(self, ipv4: str):
-        try:
-            info = self.reader.get(ipv4)
-        except ValueError:
-            raise falcon.HTTPBadRequest('message', 'Invalid IPv4 address')
-
-        result = {
-            'ip' : ipv4,
-            'continent_code' : '',
-            'continent_name' : '',
-            'country_code' : '',
-            'country_name' : '',
-            'region_code' : '',
-            'region_name' : '',
-            'city' : '',
-            'zip' : '',
-            'latitude' : '',
-            'longitude' : '',
-            'location' : {
-                'geonameid' : '',
-                'time_zone' : ''
-            }
-        }
-
-        if 'subdivisions' in info:
-            result['region_code'] = info['subdivisions'][0]['iso_code']
-            result['region_name'] = info['subdivisions'][0]['names']['en']
-
-        if 'city' in info:
-            result['city'] = info['city']['names']['en']
-            result['location']['geonameid'] = info['city']['geoname_id']
-
-        if 'postal' in info:
-            result['zip'] = info['postal']['code']
-
-        if 'continent' in info:
-            result['continent_code'] = info['continent']['code']
-            result['continent_name'] = info['continent']['names']['en']
-
-        if 'country' in info:
-            result['country_code'] = info['country']['iso_code']
-            result['country_name'] = info['country']['names']['en']
-
-        if 'location' in info:
-            result['latitude'] = info['location']['latitude']
-            result['longitude'] = info['location']['longitude']
-            result['location']['time_zone'] = info['location']['time_zone']
+        result['ip'] = ipv4
+        if GEOIP_SERVICE == 1:
```

Comment:
It looks subsequent statements are also individual based on GEOIP_SERVICE. How about making 2 methods for each this GEOIP_SERVICE?
Let's say somethinkg like below 🤔 
This is just an idea though.

```
def get(self, ipv4: str):
    if GEOIP_SERVICE == 1:
        result = self.get_geoip_service_1()
    else:
        result = self.get_geoip_service_2()
     
    return result
```

Reply:
okay ill refactor this this sounds good